### PR TITLE
dai: change to use new version of dai_config_get

### PR DESCRIPTION
**Pull request by juimonen**
Start using new version of dai_config_get where config struct is given as pointer argument from outside.

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>

### CODE REVIEW

**Comment by juimonen** [comment]
depends on zephyr side: https://github.com/zephyrproject-rtos/zephyr/pull/53708

**Review by lgirdwood — APPROVED** [review]
@abonislawski pls review

File: src/lib/dai.c
Comment (ranj063) [inline src/lib/dai.c:160]: @juimonen is this correct? Should you continue here instead?

Comment (juimonen) [reply src/lib/dai.c:160]: I mean dai_config_get should return 0 on success, otherwise negative. You kind of always get some config back, it is just testing the arguments you give are not bogus... so in that sense this can't fail, just added the check for completeness.

Comment (ranj063) [reply src/lib/dai.c:160]: even if this is not possible in practice, the right thing to do here is continue instead of return NULL right?

Comment (juimonen) [reply src/lib/dai.c:160]: yes you're right.

**Review by ranj063 — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@juimonen some conflicts.

**Comment by kv2019i** [comment]
@juimonen Would be good to add some tag/note that this cannot be merged until the Zephyr dependency is merged (and a link to it).

**Review by tmleman — APPROVED** [review]

**Comment by lgirdwood** [comment]
@juimonen any ETA for the Zephyr PR ?

**Comment by juimonen** [comment]
zephyr side merged. Updated/rebased for sof main and added west update for zephyr main/head.

**Comment by juimonen** [comment]
hmm, seems we are failing into a watchdog warning (treated as error), this commit is before set_config change, so can't really roll back. @softwarecki fyi.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Need to be merged together with Zephyr baseline update.

File: src/audio/dai-zephyr.c
Comment (juimonen) [reply src/audio/dai-zephyr.c]: ok updated.

**Comment by kv2019i** [comment]
So Intel System/merge/build passes (https://github.com/thesofproject/sof/issues/7132 is NOT hit which is surprising), but the SOF CI fails due to the DMA API changes https://sof-ci.01.org/sofpr/PR6939/build4024/devicetest/index.html

**Review by kv2019i — APPROVED** [review]
Good to go now, let's wait until CI finishes.

**Comment by kv2019i** [comment]
One failure in CI https://sof-ci.01.org/sofpr/PR6939/build4057/devicetest/index.html , but no errors in the FW logs.

**Review by abonislawski — APPROVED** [review]

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
> I also think that good idea is to point the zephyr Pull Request in the west manifest. This way the code will compile and pass validation.

Exactly that. This PR should have been combined with a manifest update. Instead it broke the daily tests ( https://github.com/thesofproject/sof/actions/runs/4228168969/jobs/7343348093) and caused very confusing PR results (e.g. https://github.com/thesofproject/sof/actions/runs/4233634492/jobs/7354897229)

**Comment by marc-hb** [comment]
Daily compilation is green again https://github.com/thesofproject/sof/actions/workflows/daily-tests.yml

**Comment by kv2019i** [comment]
@marc-hb wrote:
> Exactly that. This PR should have been combined with a manifest update. Instead it broke the daily tests ( 
> https://github.com/thesofproject/sof/actions/runs/4228168969/jobs/7343348093) and caused very confusing PR results (e.g.
> https://github.com/thesofproject/sof/actions/runs/4233634492/jobs/7354897229)

? This pr WAS combined with a manifest update, in a single git commit, just because of this.

**Comment by marc-hb** [comment]
Then why did daily tests fail? (for just one day)

https://github.com/thesofproject/sof/actions/runs/4228168969/jobs/7343348093

**Comment by marc-hb** [comment]
My bad, daily tests failed only when NOT using the manifest but the latest Zephyr. Sorry for the noise.

@aborisovich I cannot see the warning in the Windows build,
https://github.com/thesofproject/sof/actions/runs/4228168969/jobs/7343348530
This is suspicious because it's the same toolchain producing the same binaries, compared every time with Linux.

**Pull request merged**
